### PR TITLE
Stats: Change default amount of points for each period

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to #83498

## Proposed Changes

* avoid defaulting to 30 days for all available periods
* show more data points if a period other than `days` was selected before interacting with the date picker
* interacting with the date picker still applies a start date to the URL and ignores the default values mentioned above

## Testing Instructions

* go to Stats on the live brach
* without interacting with the date picker change chart's period from the dropdown below the picker
* check all possible intervals and make sure that in neither case you see a single bar on the chart
* start interacting with the date picker and verify that it works as expected

| Before | After |
| --- | --- |
|  SCR-20231026-mlzb | SCR-20231026-mlrj |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/site.jsx [hunk 1]
```diff
@@ -233,26 +251,33 @@ class StatsSite extends Component {
...
 				customChartRange.chartStart = moment()
 					.subtract( daysInRange, 'days' )
 					.format( 'YYYY-MM-DD' );
 			}
-			this.state.customChartQuantity = quantityForDaysAndPeriod( daysInRange, period );
-			customChartRange.daysInRange = daysInRange;
+
+			// how many units of data to ask for the bar for a given period
+			this.state.customChartQuantity = convertDaysToPeriodUnits( daysInRange, period );
+			customChartRange.daysInRange = daysInRange; // never used
```
Comment (grzegorz-cp) [inline client/my-sites/stats/site.jsx:280]: @a8ck3n do you recall if this is used anywhere? I can't find any reference at the moment.

Comment (a8ck3n) [reply client/my-sites/stats/site.jsx:280]: It's used when matching the current shortcut here:

https://github.com/Automattic/wp-calypso/blob/a5bd4ee8e3f589419c6cf9ad6e3d11a36b0cb992/client/components/stats-date-control/index.tsx#L106

File: client/my-sites/stats/site.jsx [hunk 2]
```diff
@@ -200,6 +201,23 @@ class StatsSite extends Component {
 		return isValid ? inputDate : null;
 	}
 
+	// Return a default amount of days to subtracts from the present day depending on the period selected.
+	// Used in case no starting date is present in the URL.
+	getDefaultRangeForPeriod( period ) {
```
Comment (a8ck3n) [inline client/my-sites/stats/site.jsx:206]: Maybe `getDefaultDaysForPeriod()`? My initial reading is that it's returning a "range" which I'd expect to be two values (or an object).

File: client/my-sites/stats/site.jsx [hunk 3]
```diff
@@ -233,26 +251,33 @@ class StatsSite extends Component {
...
+
 			if ( chartEnd ) {
 				customChartRange = { chartEnd };
 			} else {
 				customChartRange = { chartEnd: moment().format( 'YYYY-MM-DD' ) };
 			}
+
 			// Sort out quantity for chart. Default to 30 days.
-			let daysInRange = 30;
+			let daysInRange = this.getDefaultRangeForPeriod( period );
```
Comment (a8ck3n) [inline client/my-sites/stats/site.jsx:262]: Should update or remove the comment to keep it consistent with the code. Probably not needed now that we have a descriptive function name in place.

**Review by a8ck3n — APPROVED** [review]
Very nice! I much prefer these defaults for the chart to the previous cases where a single column is drawn. The only issue I'm seeing is that the chart doesn't always represent the calculated range. Try the following:

1. Visit the Stats → Traffic page.
2. Manually edit the URL and set the period to `month`.
3. Refresh the page.

Depending on the current date, it's very likely you'll see the chart and range get out of sync per this screenshot:

SCR-20231026-nslz

I manually accounted for this on [line 132](https://github.com/Automattic/wp-calypso/blob/88dffe8b81e0fe73024ade3b4cf0b6b749eaf047/client/my-sites/stats/site.jsx#L132) but it looks like that logic should apply to months and weeks too.

**Review by kangzj — APPROVED** [review]
Thanks for fixing this. The changes look reasonable to me. I'm happy for the PR to merge and follow up with Kevin's suggestions. 👍

**Pull request merged**